lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -71,7 +71,11 @@ func (handler *ContainersHandlersImpl) CreateHandler(params containers.CreatePar
 	log.Debugf("Env: %#v", params.CreateConfig.Env)
 	log.Debugf("WorkingDir: %#v", params.CreateConfig.WorkingDir)
 
-	id := exec.GenerateID().String()
+	containerID := exec.GenerateID()
+	id := containerID.String()
+
+	// inject hostname into environment
+	params.CreateConfig.Env = append(params.CreateConfig.Env, fmt.Sprintf("HOSTNAME=%s", containerID.TruncateID()))
```

Comment:
I think we need to move this into  container.go, next to this guy https://github.com/vmware/vic/blob/master/lib/apiservers/engine/backends/container.go#L172